src/core/qgsstoredexpressionmanager.h

```diff
@@ -0,0 +1,158 @@
+                          qgsstoredexpressionmanager.h
+                             -------------------
+    begin                : August 2019
+    copyright            : (C) 2019 David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSTOREDEXPRESSIONMANAGER_H
+#define QGSSTOREDEXPRESSIONMANAGER_H
+
+#include "qgis_core.h"
+#include <QString>
+#include <QObject>
+#include <QUuid>
+
+
+#ifdef SIP_RUN
+% ModuleHeaderCode
+#include <qgsstoredexpressionmanager.h>
+% End
+#endif
+
+class QDomNode;
+class QDomDocument;
+
+/**
+ * \ingroup core
+ * Stored expression containing name, content (expression text) and a category tag.
+ * \since QGIS 3.10
+ */
+struct CORE_EXPORT QgsStoredExpression
+{
+
+  /**
+   * Categories of use cases
+   * FilterExpression for stored expressions to filter attribute table
+   */
+  enum Category
+  {
+    FilterExpression = 1 << 0 //!< Expressions to filter features
+  };
+
+#ifndef SIP_RUN
+  QgsStoredExpression() = default;
+
+  QgsStoredExpression( QString name, QString expression, Category tag = Category::FilterExpression )
+    : id( QUuid::createUuid().toString() ),
+      name( name ),
+      expression( expression ),
+      tag( tag )
+  {}
+#endif
+
+  QString id;
+  QString name;
+  QString expression;
+  Category tag;
+};
+
+/**
+ * \ingroup core
+ * Manages stored expressions regarding creation, modification and storing in the project
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsStoredExpressionManager : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    QgsStoredExpressionManager()
+    {}
+
+    /**
+    * Adds an expression to the list
+    *
+    *  \param name              optional name of the expression
+    *  \param expression        expression text
+    *  \param tag               category of the expression
+    *  \returns generated id as QString
+    */
+    QString addStoredExpression( const QString &name, const QString &expression, const QgsStoredExpression::Category &tag =  QgsStoredExpression::Category::FilterExpression );
+
+    /**
+    * Removes an expression to the list
+    *
+    *  \param id                id of the expression as identification
+    */
+    void removeStoredExpression( const QString &id );
+
+    /**
+    * Updates an expression by id
```

Comment:
```suggestion
    * Updates an expression by \a id.
```

Reply:
We tag the parameter names in the description, even when we have it described below as `\param`?

Reply:
Yes.